pkg/gofr/datasource/file/ftp/fs_test.go

```diff
@@ -556,30 +569,346 @@ func TestRemoveDir(t *testing.T) {
 	fs := &fileSystem{
 		conn: mockFtpConn,
 		config: &Config{
-			Host:      "ftp.example.com",
-			User:      "username",
-			Password:  "password",
-			Port:      21,
-			RemoteDir: "/ftp/one",
+			Host:     "ftp.example.com",
+			User:     "username",
+			Password: "password",
+			Port:     21,
 		},
 		logger:  mockLogger,
 		metrics: mockMetrics,
 	}
+
 	// mocked logs
 	mockLogger.EXPECT().Logf(gomock.Any(), gomock.Any()).AnyTimes()
 	mockLogger.EXPECT().Errorf(gomock.Any(), gomock.Any()).AnyTimes()
 	mockLogger.EXPECT().Debug(gomock.Any()).AnyTimes()
-	mockMetrics.EXPECT().RecordHistogram(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			path := fmt.Sprintf("%v/%v", tt.basePath, tt.removePath)
+			fs.config.RemoteDir = tt.basePath
 
-			tt.mockRemoveExpect(mockFtpConn, path)
+			tt.mockRemoveExpect(mockFtpConn, tt.basePath, tt.removePath)
 
 			err := fs.RemoveAll(tt.removePath)
 
 			assert.Equal(t, tt.expectError, err != nil, tt.name)
```

Comment:
This should provide a better error reporting if a test fails

```suggestion
			if tt.expectError {
				assert.Error(t, err)
 			} else {
 				assert.NoError(t, err)
 			}
```